src/app/clusters/scenes-server/SceneTableImpl.cpp

```diff
@@ -209,7 +209,8 @@ struct SceneTableData : public SceneTableEntry, PersistentData<kPersistentSceneB
 
     CHIP_ERROR Serialize(TLV::TLVWriter & writer) const override
     {
-        CharSpan nameSpan(mStorageData.mName, mStorageData.mNameLength);
+        // TODO: if we have mNameLength, should this bin ByteSpan instead?
+        CharSpan nameSpan(mStorageData.mName, strnlen(mStorageData.mName, mStorageData.mNameLength));
```

Comment:
Please do not assume null termination. The issue is the mNameLength not managed properly. See my comment in SceneTable.h.

Reply:
I am not sure what to do here ... mNameLength seems to be potentially a lie. I agree that fixing logic would be correct, however I do not want to own memory storage logic here ... fixing properly should be a followup (hence my TODO). For now I am saying: if you lie about the length, I will truncate it at the first 0.

Reply:
@lpbeliveau-silabs FYI

Reply:
Reverted but kept todo. If you have length, it seems you support embedded nulls so maybe binary is better. Leaving up to Scene devels to figure it out.